pkg/controller/telemetry/metric.go

```diff
@@ -577,14 +577,22 @@ func (m *MetricController) updatePrometheusMetric() {
 	// delete metrics
 	workloadDeleteLength := len(deleteWorkload)
 	serviceDeleteLength := len(deleteService)
+	// Creating a copy reduces the amount of time spent adding locks in the programme
+	workloadLock.Lock()
```

Comment:
I donot think this is enough, `DeleteWorkloadMetric` function deleteWorkload is not locked.

In addition to that, can you share one lock for both servce and workload